roomba/main.c

```diff
@@ -0,0 +1,173 @@
+#include <stdbool.h>
+#include "kipr/wombat.h"
+
+void moveservo(int port, int position){
+    enable_servos();
+    set_servo_position(port, position);
+    msleep(1);
+}
+
+void setupsensordata(){
+    set_create_distance(0);
+    set_create_total_angle(0);
+}
+
+
+void printsensordata(){
+    printf("%d,%d,%s",get_create_rwdrop(),0,"right wheel drop \n");
+    printf("%d,%d,%s",get_create_rcliff(),0,"right cliff \n");
+    printf("%d,%d,%s",get_create_rlightbump(),0,"right light bumper \n");
+    printf("%d,%d,%s",get_create_rbump(),0,"right bumper \n");
+    printf("%d,%d,%s",get_create_rflightbump(),0,"right light front bumper \n");
+    printf("%d,%d,%s",get_create_rfcliff(),0,"right cliff \n");
+    printf("%d,%d,%s",get_create_rclightbump(),0,"right cliff light bumper \n");
+    printf("%d,%d,%s",get_create_cwdrop(),0,"front drop wheel \n");
+    printf("%d,%d,%s",get_create_lclightbump(),0,"left cliff light bumper \n");
+    printf("%d,%d,%s",get_create_lfcliff(),0,"left front cliff \n");
+    printf("%d,%d,%s",get_create_lflightbump(),0,"left light bumper \n");
+    printf("%d,%d,%s",get_create_lbump(),0,"left bumper \n");
+    printf("%d,%d,%s",get_create_llightbump(),0,"left light bumper \n");
+    printf("%d,%d,%s",get_create_lcliff(),0,"left cliff \n");
+    printf("%d,%d,%s",get_create_lwdrop(),0,"left wheel drop \n");
+    printf("%d,%d,%s",get_create_distance(),0,"total distance\n");
+    printf("%d,%d,%s",get_create_total_angle(),0,"total angle \n");
+}
+
+bool setupcamera(){
+    if(camera_open() == 0){
+        return false;
+    }
+    if(camera_update() == 0)
+    {
+        return false;
+    }
+    return true;
+}
+
+void setuprobot(bool debug, bool camera){
+    create_connect();
+    setupsensordata();
+    if(debug == true){
```

Comment:
Define Debug-Specific Code with C Preprocessor operators!

https://learn.microsoft.com/en-us/cpp/preprocessor/hash-if-hash-elif-hash-else-and-hash-endif-directives-c-cpp?view=msvc-170